### PR TITLE
Enable POST requests to /_changes in clustered CouchDB

### PR DESCRIPTION
**Pull request by willholley**
Appropriate handling of a POST to _changes is already in place in single node CouchDB. In clustered CouchDB, allow POST requests to pass through to the underlying handler and existing logic will do the rest.

COUCHDB-2530

### CODE REVIEW

File: src/chttpd_db.erl [hunk 1]
```diff
@@ -61,7 +61,7 @@ handle_request(#httpd{path_parts=[DbName|RestParts],method=Method,
         do_db_req(Req, Handler)
     end.
 
-handle_changes_req(#httpd{method='GET'}=Req, Db) ->
+handle_changes_req1(#httpd{}=Req, Db) ->
```
Comment (kxepal) [inline src/chttpd_db.erl]: `handle_changes_req1` should comes after `handle_changes_req` functions. You should be able to read the code from top to bottom, not in reverse.

File: src/chttpd_db.erl [hunk 2]
```diff
@@ -87,7 +87,13 @@ handle_changes_req(#httpd{method='GET'}=Req, Db) ->
...
-    end;
+    end.
+
+handle_changes_req(#httpd{method='POST'}=Req, Db) ->
+    couch_httpd:validate_ctype(Req, "application/json"),
+    handle_changes_req1(Req, Db);
+handle_changes_req(#httpd{method='GET'}=Req, Db) ->
+    handle_changes_req1(Req, Db);
 handle_changes_req(#httpd{path_parts=[_,<<"_changes">>]}=Req, _Db) ->
     send_method_not_allowed(Req, "GET,HEAD").
```
Comment (kxepal) [inline src/chttpd_db.erl]: You forgot to define `POST` as allowed method here.

**Comment by willholley** [comment]
@kxepal thanks - addressed both points in the latest push

**Comment by kxepal** [comment]
Cool, LGFM.

**Comment by kxepal** [comment]
Landed, thank you! (:

**Pull request closed without merging**